twill-api/src/main/java/org/apache/twill/api/TwillPreparer.java

```diff
@@ -227,15 +227,36 @@
   TwillPreparer addSecureStore(SecureStore secureStore);
 
   /**
-   * Set the log level for Twill applications running in a container.
+   * Set the root log level for Twill applications in all containers.
    *
    * @param logLevel the {@link LogEntry.Level} that should be set.
    *                 The level match the {@code Logback} levels.
    * @return This {@link TwillPreparer}.
+   * @deprecated Use {@link #setLogLevels(Map)} with key {@link org.slf4j.Logger#ROOT_LOGGER_NAME} instead.
    */
+  @Deprecated
   TwillPreparer setLogLevel(LogEntry.Level logLevel);
 
   /**
+   * Set the log levels for requested logger names for Twill applications running in a container. The log level whose
+   * value is null will be ignored
```

Comment:
I think the implementation actually throws `IllegalArgumentException`, which is better than silently ignored. Please update the doc here accordingly.